lib/api/controllers/controller.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const
+  {
+    errors: {
+      InternalError: KuzzleInternalError
+    }
+  } = require('kuzzle-common-objects');
+
+// Base class for all controllers (except the funnel one)
+class BaseController {
+  constructor(kuzzle) {
+    this.kuzzle = kuzzle;
+  }
+
+  /**
+   * Check if the provided action name exists within that controller.
+   * This check's purpose is to prevent actions leak by making actions exposure
+   * explicit.
+   *
+   * @param  {string} name
+   * @return {boolean}
+   */
+  isAction(name) {
+    throw new KuzzleInternalError(
+      `Call to incomplete controller when invoking the action ${name}`);
```

Comment:
Why this method has to be re-implemented for each controller ? It can just check the `actions` property content so we can avoid code duplication in every controllers

Reply:
Resolving: same as the previous comment.